src/data/store/WfsFeatures.js

```diff
@@ -0,0 +1,250 @@
+     * @cfg {String}
+     */
+    count: null,
+
+    /**
+     * Offset to add to the #startIndex in the WFS request.
+     * @cfg {Number}
+     */
+    startIndexOffset: 0,
+
+    /**
+     * The OL format used to parse the WFS GetFeature response.
+     * @cfg {ol.format.Feature}
+     */
+    format: null,
+
+    /**
+     * The attribution added to the created vector layer source. Only has an
+     * effect if #createLayer is set to `true`
+     * @cfg {String}
+     */
+    layerAttribution: null,
+
+    /**
+     * Constructs the WFS feature store.
+     *
+     * @param {Object} config The configuration object.
+     * @private
+     */
+    constructor: function(config) {
+        var me = this;
+
+        config = config || {};
+
+        // apply count as store's pageSize
+        config.pageSize = config.count || me.count;
+
+        if (config.pageSize > 0) {
+            // calculate initial page
+            var startIndex = config.startIndex || me.startIndex;
+            var currentPage = Math.floor(startIndex / config.pageSize) + 1;
+            config.currentPage = currentPage;
+        }
+
+        // avoid creation of vector layer by parent class (raises error when
+        // applying WFS data) so we can create the WFS vector layer on our own
+        // (if needed)
+        var createLayer = config.createLayer;
+        config.createLayer = false;
+
+        me.callParent([config]);
+
+        if (!me.url) {
+            Ext.raise('No URL given to WfsFeaturesStore');
+        }
+
+        if (createLayer) {
+            // the WFS vector layer showing the WFS features on the map
+            me.source = new ol.source.Vector({
+                features: [],
+                attributions: me.layerAttribution
+            });
+            me.layer = new ol.layer.Vector({
+                source: me.source,
+                style: me.style
+            });
+
+            me.layerCreated = true;
+        }
+
+        // initally load the WFS data
+        me.loadWfs();
+
+        // before the store gets re-loaded (e.g. by a paging toolbar) we trigger
+        // the re-loading of the WFS, so the data keeps in sync
+        me.on('beforeload', me.loadWfs, me);
+
+        // add layer to connected map, if available
+        if (me.map && me.layer) {
+            me.map.addLayer(me.layer);
+        }
+    },
+
+    /**
+     * Detects the total amount of features (without paging) of the given
+     * WFS response. The detectioin is based on the response format (currently
+     * GeoJSON and GML >=v3 are supported).
+     *
+     * @private
+     * @param  {Object} wfsResponse The XMLHttpRequest object
+     * @return {Integer}            Total amount of features
+     */
+    getTotalFeatureCount: function(wfsResponse) {
+        var me = this;
+        var totalCount = -1;
+        if (me.outputFormat.indexOf('application/json') !== -1) {
+            var respJson = Ext.decode(wfsResponse.responseText);
```

Comment:
This can still be an XML, right? @weskamm said this would blow up then... maybe `try {} catch (e) {}`?

Reply:
Correct. Since errors in OGC services are encoded in XML, no matter what you send as outputFormat (:-/), this could blow up. I added a try / catch in 0aa4b81